src/pymor/algorithms/timestepping.py

```diff
@@ -10,227 +10,619 @@
+from pymor.parameters.base import Mu
+from pymor.tools import floatcmp
+from pymor.tools.deprecated import Deprecated
 from pymor.vectorarrays.interface import VectorArray
 
 
 class TimeStepper(ImmutableObject):
     """Interface for time-stepping algorithms.
 
-    Algorithms implementing this interface solve time-dependent problems
+    Algorithms implementing this interface solve time-dependent initial value problems
     of the form ::
 
-        M * d_t u + A(u, mu, t) = F(mu, t).
+        M * d_t u + A(u, mu, t) = F(mu, t),
+                     u(mu, t_0) = u_0(mu).
+
+    Time-steppers used by |InstationaryModel| have to fulfill this interface. Time evolution can be performed
+    by calling :meth:`solve`.
 
-    Time-steppers used by |InstationaryModel| have to fulfill
-    this interface.
+    Note that the actual work is done in an iterator derived from :class:`TimeStepperIterator`.
+
+    Parameters
+    ----------
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each intermediate vector that is
+        calculated is returned. Else an interpolation of the calculated vectors on an equidistant temporal grid is
+        returned, using an appropriate interpolation of the respective time stepper.
+    interpolation
+        Type of temporal interpolation to be used. Currently implemented are: piecewise constant (P0) and piecewise
+        linear (P1).
     """
 
-    @abstractmethod
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
+    IteratorType = None
+    available_interpolations = ('P0', 'P1')
+
+    def __init__(self, num_values=None, interpolation='P1'):
+        assert not num_values or (isinstance(num_values, Number) and num_values > 1)
+        assert interpolation in self.available_interpolations
+        self.__auto_init(locals())
+
+    def solve(self, t0, t1, U0, A, F=None, M=None, mu=None, return_iter=False, return_times=False):
         """Apply time-stepper to the equation ::
 
-            M * d_t u + A(u, mu, t) = F(mu, t).
+            M * d_t u + A(u, mu, t) = F(mu, t),
+                         u(mu, t_0) = u_0(mu).
 
         Parameters
         ----------
-        initial_time
+        t0
             The time at which to begin time-stepping.
-        end_time
+        t1
             The time until which to perform time-stepping.
-        initial_data
-            The solution vector at `initial_time`.
-        operator
+        U0
+            The solution vector at `t0`.
+        A
             The |Operator| A.
-        rhs
+        F
             The right-hand side F (either |VectorArray| of length 1 or |Operator| with
             `source.dim == 1`). If `None`, zero right-hand side is assumed.
-        mass
+        M
             The |Operator| M. If `None`, the identity operator is assumed.
         mu
             |Parameter values| for which `operator` and `rhs` are evaluated. The current
```

Comment:
```suggestion
            |Parameter values| for which A and F are evaluated. The current
```

Reply:
we decided to keep `operator`, `rhs`, and `mass`